### PR TITLE
added bucket sort method tutorial with two solutions

### PR DESCRIPTION
**Pull request by ARRY7686**
## Change Summary

Edited the Bucket Sort File and made solutions to new problems **(sort array by parity ii and maximum gap)**

## Checklist

**If you haven't fulfilled the below requirements or even delete the entire checklist, your PR won't be reviewed and will be closed without notice. Regular contributors (with 10+ PRs) can skip this part.**

### General 

- [x] This Pull Request is all my own work. (You'll be blacklisted if you are caught for plagiarism.)
- [x] I've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md)
- [x] I've applied LaTex for all variables, formulas and time / space complexity instead of using backticks
- [x] I've started the app locally and verified all the content and all links (if applicable) are accessible correctly
- [x] I've included Complexity Analysis (Time Complexity & Space Complexity).
- [x] I've written my explanation well and it is easy to understand for beginners

### Tutorial

- [x] I've read and followed the [Tutorial Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#tutorial-template)
- [x] I've explained my topic well with 2 - 3 LC problems and no external problems are used. 
- [x] I've provided the full working solutions to the problems used in this tutorial.
- [x] I've provided suggested problems at the end with the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [x] I've given credits / references if I use external resources. (For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).)

### Solutions

- [x] I've read and followed the [Solution Template](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md#solution-template)
- [x] I've formatted my code well with [K&R Coding style](https://gist.github.com/jesseschalken/0f47a2b5a738ced9c845#why-kr)
- [x] I've confirmed that comments are put above each line rather than writing on the same line.
- [x] I've included a meaningful approach name for my solution. e.g. `## Approach 1: Two Pointers`.

### CODE REVIEW

**Comment by ARRY7686** [comment]
this solves issue #89

**Comment by ARRY7686** [comment]
@wingkwong I have solved this issue pls mark it as hacktober fest before merging it

File: tutorials/basic-topics/sorting/bucket-sort.md [hunk 1]
```diff
@@ -11,4 +10,131 @@ keywords:
...
+```
+</TabItem>
+</Tabs>
+
+export const suggestedProblems = [
+  {
+    "problemName": "0075 -  Sort Array By Parity II ",
+    "difficulty": "Medium",
+    "leetCodeLink": "https://leetcode.com/problems/sort-colors/",
+    "solutionLink": "../../../solutions/0900-0999/0922-Sort-Array-By-Parity-ii-easy"
```
Comment (wingkwong) [inline tutorials/basic-topics/sorting/bucket-sort.md:130]: please double check

**Review by wingkwong — CHANGES_REQUESTED** [review]
1. not detailed enough
2. please use LC problems to walk through the algorithm. not just pasting at the end.

**Comment by ARRY7686** [comment]
@wingkwong resolved the issue kindly check

**Review by wingkwong — CHANGES_REQUESTED** [review]
1. please check other tutorials to see the format. 
2. I'm expecting 2-3 LC problems in Algorithm while explaining the algo. The target audience is beginners. I would expect step by step explanations. 
3. dry run only explains the code, not how the algo solves it. i.e. i can just read the code without reading the tutorial. it should link to what you wrote in Algorithm section.
4. same comment on suggested problems. the link and problem name don't match

**Comment by wingkwong** [comment]
Closing due to inactivity

**Pull request closed without merging**